blueocean-dashboard/src/main/js/components/RunMessageCell.jsx

```diff
@@ -25,22 +28,73 @@ export default class RunMessageCell extends Component {
         const showCauses = run && (run.causes && run.causes.length > 0);
 
         if (showUserDefinedMessage) {
-            message = (<span className="RunMessageCell" title={run.description}><span className="RunMessageCellInner">{run.description}</span></span>);
+            message = (
+                <span className="RunMessageCell" title={run.description}>
+                    <span className="RunMessageCellInner">
+                        <Link to={linkTo} className="unstyled-link" >
+                            {run.description}
+                        </Link>
+                    </span>
+                </span>
+            );
         } else if (showCommitMessage) {
-            const commitMsg = run.changeSet[run.changeSet.length - 1].msg;
+            let commitMsg = run.changeSet[run.changeSet.length - 1].msg;
+            const commitMsgWithIssues = [];
+            const linkedCommitMsg = (<Link to={linkTo} className="unstyled-link" >{commitMsg}</Link>);
+
+            if (run.changeSet[run.changeSet.length - 1].issues) {
+                let issuesIdString = '';
+                const issuesObj = {};
+
+                for (const issue of run.changeSet[run.changeSet.length - 1].issues) {
+                    issuesIdString += `${issue.id}|`;
+                    issuesObj[issue.id] = issue.url;
+                }
+
+                if (issuesIdString) {
```

Comment:
Is this block of code duplicating the functionality of the `AddIssuesLinksToMsg` component?

Reply:
@sophistifunk I just did a project wide search of AddIssuesLinksToMsg and got no results, can you point me in the right direction to find this component? I have no idea what it does.

Reply:
Unless Github is having a stroke, you added it to RunDetailsChanges in this PR?

Reply:
argh, my bad, I was on the wrong branch and didn't realise, there is a difference between the two codes, the one in RunMessageCell adds a link to the text that is not a jira issue, in RunDetailsChanges the AddIssuesLinksToMsg component only adds the jira links to a text

Reply:
Righto, just wanted to check as it seemed pretty similar :)

Reply:
I preferred to have them separate because the code is pretty complicated as is and if I make the component take another optional link prop for the text that is not a jira issue it would add another bit of complexity that would make it even harder to maintain, what do you think, should I make AddIssuesLinksToMsg a separate component that takes in that optional prop and use it in both use cases?

Reply:
I am wondering if we could solve this by having a more general purpose component called `LinkifiedText` which would take in two props:
1. `text:String` as the string of text (commit message, or whatever)
2. `links:Map` where the key is the text that should be linked, and the value is the absolute/relative URL. 

If there are no links, the component could just return a `span` with the `text` as its child. That component wouldn't be too hard to impl and could have a unit test or two.

Then each part of the UI that needs linkified text just implements a simple pure function that takes in the data of interest (commit, changeSet, etc) and builds the `links` Map. If `links` is empty, then we know there are no links and we can fall back to the normal linking behavior.

Note: we don't need to use an actual ES Map impl, just an object literal would be fine.

I may have missed a subtle difference but if so, please clarify and we can find the best way to resolve it.